model/patch/patch_test.go

```diff
@@ -946,19 +946,23 @@ func TestSetChildPatches(t *testing.T) {
 }
 
 func TestGetCollectiveStatusFromPatchStatuses(t *testing.T) {
-	successful := []string{evergreen.LegacyPatchSucceeded}
-	assert.Equal(t, evergreen.LegacyPatchSucceeded, GetCollectiveStatusFromPatchStatuses(successful))
+	successful := []string{evergreen.VersionSucceeded}
+	assert.Equal(t, evergreen.VersionSucceeded, GetCollectiveStatusFromPatchStatuses(successful))
```

Comment:
Since `GetCollectiveStatusFromPatchStatuses` is used in the REST API to [decide the API model patch's status](https://github.com/evergreen-ci/evergreen/blob/dc44657541b9a8eb6b746bd03679cde291061b13/rest/model/patch.go#L319), it seems like this will change the output from the REST API from "succeeded" to "success". I wonder if people have existing scripts (or maybe use libraries like [evergreen.py](https://github.com/evergreen-ci/evergreen.py/blob/d989b8f84dd370ee42315256ed95dca100d12c78/tests/evergreen/data/patch.json#L11)) and are still looking for "succeeded" but aren't aware that this is about to change to "success" - is it worth doing any comms to raise awareness?

Reply:
Yeaaaahh we should either do some sort of comms, or we could translate this back to "succeeded" for the rest response 🤔

Reply:
@Kimchelly re-requesting review -- with the assumption that we're going to do comms, could you review the changes?

Reply:
I added something to ensure that API response is consistently the new thing but if we're concerned we could change it to be consistently the old thing.